### PR TITLE
Async time coordinator

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will add an async time coordinator and rework the flag definitions to be a little clearer

### CODE REVIEW

File: src/helics/core/AsyncTimeCoordinator.hpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+#include "json/forwards.h"
+#include <atomic>
+#include <functional>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace helics {
+
+/** class implementing no time coordination
```
[the remark this answers is not in the document]
Comment (nightlark) [reply src/helics/core/AsyncTimeCoordinator.hpp:23]: I think more explanation of what it means to be an Async time coordinator and how to use it is needed -- if it actually does nothing, then something like NullTimeCoordinator seems like a more accurate name.

File: src/helics/core/BrokerBase.cpp [hunk 2]
```diff
@@ -150,6 +151,24 @@ std::shared_ptr<helicsCLI11App> BrokerBase::generateBaseCLI()
         "--globaltime",
         globalTime,
         "specify that the broker should use a globalTime coordinator to coordinate a master clock time with all federates");
+    hApp->add_flag(
+        "--async",
+        asyncTime,
+        "specify that the federation should not synchronize time (time management is done outside of helics or not relevant");
```
Comment (nightlark) [inline src/helics/core/BrokerBase.cpp:157]: I think I'm having the same trouble with this description as the class description -- `AsyncTimeCoordinator` feels like it implies some different time coordination is still happening. It sounds like this option is really intended for mechanisms outside HELICS to provide time management, so maybe `ExternalTimeCoordinator` would match better with what it does and how it is used.

File: src/helics/core/BrokerBase.cpp [hunk 3]
```diff
@@ -150,6 +151,24 @@ std::shared_ptr<helicsCLI11App> BrokerBase::generateBaseCLI()
...
+                    asyncTime = true;
+                } else if (arg == "global") {
+                    globalTime = true;
+                } else {
+                    asyncTime = false;
+                    globalTime = false;
+                }
+            },
+            "specify the timing method to use in the broker")
+        ->check(CLI::IsMember({"async", "global", "distributed"}));
```
Comment (nightlark) [inline src/helics/core/BrokerBase.cpp:171]:
```suggestion
        ->check(CLI::IsMember({"async", "global", "distributed", "default"}));
```

Documentation lists "default" as one of the values.

File: src/helics/core/BrokerBase.cpp [hunk 4]
```diff
@@ -150,6 +151,24 @@ std::shared_ptr<helicsCLI11App> BrokerBase::generateBaseCLI()
         "--globaltime",
         globalTime,
         "specify that the broker should use a globalTime coordinator to coordinate a master clock time with all federates");
+    hApp->add_flag(
+        "--async",
```
Comment (nightlark) [inline src/helics/core/BrokerBase.cpp:155]:
```suggestion
        "--asynctime",
```

Matches the `globaltime` option by adding a `time` suffix. Might also make it clearer that the option is related to time coordination and not something else like communication.

File: src/helics/core/AsyncTimeCoordinator.hpp [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+#include <utility>
+#include <vector>
+
+namespace helics {
+
+/** class implementing a time coordinate that explicitly allowed asynchronous operation of the
+federates.  That is the federates are not time synchronized and are allowed to operate ahead of
+others potential uses are 1). The use of an external time coordination mechanism. 2). A purely
+command driven system 3).  All federates are operated in real time mode 4).  You don't care about
+the data and just want to see how fast all the federates go on their own.
```
Comment (nightlark) [inline src/helics/core/AsyncTimeCoordinator.hpp:27]:
```suggestion
/** class implementing a time coordinator that explicitly allows asynchronous operation of the
federates -- that is, the federates are not time synchronized and are allowed to operate ahead of
others. Potential uses are: 1) the use of an external time coordination mechanism, 2) a purely
command driven system, 3) all federates are operated in real time mode, 4) you don't care about
the data and just want to see how fast all the federates go on their own.
```

File: docs/references/configuration_options_reference.md [hunk 6]
```diff
@@ -295,6 +295,9 @@ String used to define the configuration of the broker if one is autogenerated. S
 - `--file_log_level=` - Specifies the level of logging to file for this broker.
 - `--console_log_level=` - Specifies the level of logging to file for this broker.
 - `--dumplog` - Captures a record of all logging messages and writes them out to file or console when the broker terminates.
+- `--globaltime` - Specify that the broker should use a globalTime coordinator to coordinate a master clock time with all federates.
+- `--async` - Specify that the federation should use the asynchronous time coordinator (only minimal time management is handled in HELICS and federates are allowed to operate independently).
```
Comment (phlptp) [inline docs/references/configuration_options_reference.md:299]:
```suggestion
- `--asynctime` - Specify that the federation should use the asynchronous time coordinator (only minimal time management is handled in HELICS and federates are allowed to operate independently).
```

File: tests/helics/system_tests/TimingTests.cpp [hunk 7]
```diff
@@ -707,3 +707,78 @@ TEST_F(timing, dual_fast_sender_tests_ci_skip)  // ci_skip
 }
 
 #endif
+
+TEST_F(timing, async_timing)
+{
+    extraBrokerArgs = "--async";
```
Comment (phlptp) [inline tests/helics/system_tests/TimingTests.cpp:713]:
```suggestion
    extraBrokerArgs = "--asynctime";
```

File: tests/helics/system_tests/TimingTests.cpp [hunk 8]
```diff
@@ -707,3 +707,78 @@ TEST_F(timing, dual_fast_sender_tests_ci_skip)  // ci_skip
...
+    res = vFed2->requestTime(5.0);
+    EXPECT_LT(res, 4.0);
+
+    vFed1->finalize();
+    vFed2->finalize();
+}
+
+TEST_F(timing, async_timing_message)
+{
+    extraBrokerArgs = "--async";
```
Comment (phlptp) [inline tests/helics/system_tests/TimingTests.cpp:750]:
```suggestion
    extraBrokerArgs = "--asynctime";
```

**Review by nightlark — APPROVED** [review]

**Pull request merged**